modular/gater/object_handler.go

```diff
@@ -140,12 +140,24 @@ func (g *GateModular) getObjectHandler(w http.ResponseWriter, r *http.Request) {
 		}
 		log.CtxDebugw(reqCtx.Context(), reqCtx.String())
 	}()
-
+	
 	reqCtx, err = NewRequestContext(r)
 	if err != nil {
-		return
+		// check the object's visibility type whether equal to public read
+		authorized, err = g.baseApp.Consensus().VerifyGetObjectPermission(reqCtx.Context(), "",
```

Comment:
What if the user has GetObject permission and the object is private?

Reply:
new request context return err == nil

Reply:
Sorry, I didn't saw `VerifyPermission` in `NewRequestContext`.